codecov.yml

```diff
@@ -0,0 +1,68 @@
+codecov:
+  require_ci_to_pass: yes
+
+coverage:
+  # Do not need token here, because of the public repo here
+  # token:
+  # bot:
+  # ci:
+  # The age you want coverage reports to expire at(hours)
+  max_report_age: 24
+  # Should Codecov wait for all other statues to pass before sending it's status.
+  require_ci_to_pass: yes
+  notify:
+    # How many uploaded reports Codecov should wait to receive before sending statuses
+    after_n_builds: 2
+    # Should Codecov wait for all CI statuses to complete before sending ours.
+    wait_for_ci: yes
+
+  # What precision do you want the coverage value to be
+  precision: 2
+  # Which direction to you want to round the coverage value
+  round: down
+  range: "50...100"
+
+  status:
+    project:
+      default:
+        # basic
+        target: auto
+        # Allow the coverage to drop by X%, and posting a success status.
+        threshold: 0%
+        base: auto
+        # advanced
+        branches:
+          - master
+        if_not_found: success
+        if_ci_failed: error
+        informational: false
+        only_pulls: false
+      api:
+        target: 50%
+        flags:
+          - api
+
+flags:
+  # filer the folder(s) you wish to measure by that flag
+  api:
+    # only inlcude files in the pkg
+    paths:
+      - pkg/
```

Comment:
can this be omitted to include all paths, or do we want to list all paths explicitly?

Reply:
Nice catch, I want to list the path which includes more tool feature functions. What's your idea?

Reply:
The configuration `api` is target 50%, and the rest of file may cover by default.

Reply:
I think we could start using it to gain experience and adjust the configuration after that.

It makes sense to have different configuration for different areas of the code, depending for instance on the risk.

As I don't yet know this particular tool, I have to get familiar with it first.

We have not tests yet, so there won't be much coverage....

Reply:
Agree with you, as the repo be used to other adapters we should set the restict rules for this repo's code, and as you said above we could start use it to gain experience.